src/fstar/FStar.Interactive.Ide.fs

```diff
@@ -884,7 +615,7 @@ let collect_errors () =
   errors
 
 let run_segment (st: repl_state) (code: string) =
-  let frag = { frag_text = code; frag_line = 1; frag_col = 0 } in
+  let frag = { frag_fname = "<input>"; frag_text = code; frag_line = 1; frag_col = 0 } in
```

Comment:
Why are we passing `"<input>"` here, instead of `st.repl_fname`? Are these two difference?

Reply:
As it so happens, these are two different things. Ideally, there should be just one place where we keep track of the filename of the fragment, but this is not so in reality. In the interactive mode, there are different "modes of operation", including a mode where we send a "text fragment": in that case, the filename is `"<input>"`. In Lsp, there is a single mode of operation: we only send across fragments. To cope with this difference, we need this change.